python/ray/train/_internal/worker_group.py

```diff
@@ -86,18 +87,42 @@ def construct_metadata() -> WorkerMetadata:
     node_id = ray.get_runtime_context().get_node_id()
     node_ip = ray.util.get_node_ip_address()
     hostname = socket.gethostname()
-    gpu_ids = [str(gpu_id) for gpu_id in ray.get_gpu_ids()]
     pid = os.getpid()
 
     return WorkerMetadata(
         node_id=node_id,
         node_ip=node_ip,
         hostname=hostname,
-        gpu_ids=gpu_ids,
+        gpu_and_accelerator_ids=_get_gpu_and_accelerator_ids(),
         pid=pid,
     )
 
 
+def _get_gpu_and_accelerator_ids() -> Dict[str, List[str]]:
+    """Get GPU and accelerator IDs from runtime context for given actor/worker.
+
+    Returns:
+        A dictionary mapping resource IDs to a list of resource IDs.
+        For example,
+        {
+            "GPU": ["0", "1"],
+            "neuron_cores": ["0", "1"]
+        }
+    """
+    gpu_and_accelerator_ids = defaultdict(list)
+
+    resource_ids = ray.get_runtime_context().get_resource_ids()
+    gpu_ids = resource_ids[ray_constants.GPU]
+    neuron_core_ids = resource_ids[ray_constants.NEURON_CORES]
+
+    gpu_and_accelerator_ids[ray_constants.GPU].extend(gpu_ids)
+    gpu_and_accelerator_ids[ray_constants.NEURON_CORES].extend(neuron_core_ids)
+
+    if len(gpu_ids) > 0 and len(neuron_core_ids) > 0:
+        raise RuntimeError("Cannot support GPU and Neuron Core IDs on same Worker.")
```

Comment:
I don't think this validation should happen at this layer. The WorkerGroup is a more general utility class. I think this should either:
1. Be handled at Ray Core level if this is a fundamental problem with resource scheduling
2. Be handled at the Train Backend level if this is a problem with how the training framework supports it.

Reply:
It's handled by Ray core, removed. https://github.com/ray-project/ray/blob/89ac80d883a4361aa9a304af9c7be043129f9fb9/python/ray/_private/ray_option_utils.py#L123